typescript/sdk/src/token/EvmERC20WarpRouteReader.ts

```diff
@@ -261,87 +283,193 @@ export class EvmERC20WarpRouteReader extends HyperlaneReader {
     type: TokenType,
     warpRouteAddress: Address,
   ): Promise<HypTokenConfig> {
-    if (
-      type === TokenType.collateral ||
-      type === TokenType.collateralVault ||
-      type === TokenType.collateralVaultRebase ||
-      type === TokenType.XERC20 ||
-      type === TokenType.XERC20Lockbox
-    ) {
-      let xerc20Token: Address | undefined;
-      let lockbox: Address | undefined;
-      let token: Address;
-      let xERC20Metadata: XERC20TokenMetadata | {} = {};
-
-      if (type === TokenType.XERC20Lockbox) {
-        // XERC20Lockbox is a special case of collateral, we will fetch it from the xerc20 contract
-        const hypXERC20Lockbox = HypXERC20Lockbox__factory.connect(
-          warpRouteAddress,
-          this.provider,
-        );
-        xerc20Token = await hypXERC20Lockbox.xERC20();
-        token = xerc20Token;
-        lockbox = await hypXERC20Lockbox.lockbox();
-      } else {
-        const erc20 = HypERC20Collateral__factory.connect(
-          warpRouteAddress,
-          this.provider,
-        );
-        token = await erc20.wrappedToken();
-      }
+    // Using null instead of undefined to force
+    // a compile error when adding a new token type
+    const mapConfig: Record<
+      TokenType,
+      ((address: Address) => Promise<HypTokenConfig>) | null
```

Comment:
Would it be cleaner if the type is `(arg => result) | undefined` and we skip the entries initialized to null?

Reply:
it would have the same result as having the `| null` to avoid initialization entries that currently have null values I can wrap the record into a `Partial` and that would not require all keys of the map to be specified but that would go against my goal of having the map in the first place because i'd like users that add a new TokenType to be "notified" by the compiler that they need to implement (or not) a new token config derivation function instead of simply adding the new type and forgetting this part because there are no compile time errors only to discover that the new type is not derivable when using the cli and having an error or an incorrect data read like it was before. Wdyt? @antigremlin

Reply:
Good thinking! If the point of having `| null` is to get a warning from the compiler when we add a new element to `TokenType` but forget to add a derivation function to this map, then this typing is very useful. 🚀